default/python/AI/buildings.py

```diff
@@ -0,0 +1,48 @@
+import freeOrionAIInterface as fo
+from enum import Enum
+from typing import List
+
+from common.fo_typing import PlanetId
+from freeorion_tools.caching import cache_for_current_turn, cache_for_session
+
+
+class Building(Enum):
+    """
+    Enum to represent fo building template.
+
+    Pros:
+    - Shorten code a bit
+    - Allow to have shorter names for ids,
+      we could get read of BLD prefix, since it always use with Building. prefix.
+    - Possible to add automated check that buildings exists and all building are matched.
+
+    Cons:
+    - fo code for getting id returns strings, probably will need to make an adapters.
+
+    Thoughts:
+    - We could replace it in iterative way one constant at the time,
+      this will require to use `value` a lot. When all ids are converted we could get rid of `value` usage.
```

Comment:
Well, when we wrap building releated fo-functions (as you've done it with available and queue_in), most of the value calls will be done by the class itself.

Reply:
The idea is to use value only inside the class.  Right now we have the code that uses buildings as a string, and to change this code we should convert chunk of strings to enum.